### DISCUSSION TITLE
Post sale callback to 3rd party application

### DISCUSSION THREAD

**Discussion by pandar00** [etsy/open-api#985] [Q&A] [answered]
Use-case:
Send a customized email to buyers when a new sale is posted in the shop

Question:
What are ways to listen for new sales event so a 3rd party application can process the order in *near real-time*

Context:
My shop has gained momentum so I'm writing an automation to streamline the manual processes. Currently, I leverage both Etsy messaging and email to communicate with the buyers and get their confirmation on the design which is becoming cumbersome.

I wasn't able to find any event webhooks in Etsy similar to [Zendesk](https://support.zendesk.com/hc/en-us/articles/4408839108378-Creating-webhooks-to-interact-with-third-party-systems) or [Stripe](https://stripe.com/docs/webhooks) which means I have to poll Etsy for new updates but there's an API call limit of 10K. I need to be as real-time as possible so I'd like to poll every second but that's already 86K calls a day. Even if I relax to 1 every 10sec, that's ~9K calls which is nearly at the daily allowance.

Are there any other options?

Has anyone contacted developer@etsy.com to increase the rate limit increase? How did it go?

Thanks

**Comment by Gareth064** [comment, marked as answer]
Morning.

There are no real time feeds, webhooks or events available from the API.

Your only option is to do polling to get new orders.

You probably know this already, but it is highly recommended that you do all communication, with Etsy sellers, via the Etsy platform and not via external means like emails etc. This is to protect you, if the buyer opens a case against you. Etsy will not take into consideration any proof, during the case, which is obtained outside of their platform.

**Reply by pandar00** [reply]
> This is to protect you, if the buyer opens a case against you. Etsy will not take into consideration any proof, during the case, which is obtained outside of their platform.

I see. That's good to know. I would use Etsy messaging if there's a good way to integrate but it appears chat API isn't supported yet according to https://github.com/etsy/open-api/discussions/970 issue. Also, I ask buyers to submit a photo but some users just can't seem to figure out how to attach an image in Etsy messaging but knows well in Gmail so I had to resort to email as a secondary option for them 😅 

Anyhow, thanks for the response
